src/Interpreters/ITokenExtractor.cpp

```diff
@@ -15,10 +14,36 @@
 
 #endif
 
+namespace ErrorCodes
+{
+    extern const int NOT_IMPLEMENTED;
+}
 
 namespace DB
 {
 
+std::vector<String> ITokenExtractor::getTokens(const char * data, size_t length) const
+{
+    std::vector<String> tokens;
```

Comment:
Moved to the base class now. There was some nice .reserve logic but in practice it likely won't make a difference as the first push_back will reserve enough space anyways and we'll do exponential resize.